### PR TITLE
Properly package the canonicaljson module

### PR DESCRIPTION
**Pull request by babolivier**
Package canonicaljson as a proper module rather than a single-file module. The main motivation is to expose the `py.typed` marker file so other projects can use the typing annotations.

I've decided to go with an `src/` layout for the module's source rather than the solution proposed in https://github.com/matrix-org/python-canonicaljson/issues/50#issuecomment-1255280358 to avoid a repeat of https://github.com/matrix-org/matrix-python-common/pull/23.

Fixes https://github.com/matrix-org/python-canonicaljson/issues/50

### CODE REVIEW

File: tox.ini
Comment (DMRobertson) [inline tox.ini:14]: Why are we using two different test runners (pytest and nose2) here?

File: setup.cfg
Comment (DMRobertson) [inline setup.cfg:11]: Is there a more specific email address we can use here?

File: MANIFEST.in
Comment (DMRobertson) [inline MANIFEST.in:6]: These both seem like good things to include in the sdist. Are they included automatically now?

File: MANIFEST.in
Comment (DMRobertson) [inline MANIFEST.in:9]: Do we need *.py here?

File: MANIFEST.in
Comment (DMRobertson) [inline MANIFEST.in:2]: For the changelog?

Does the readme (RST) automatically get included?

File: MANIFEST.in
Comment (babolivier) [reply MANIFEST.in:9]: Possibly not - this bit is copied from matrix-common's manifest.

File: tox.ini
Comment (babolivier) [reply tox.ini:14]: My mistake, I went over this quickly while trying to fix other things and hadn't realised nose2 was a test runner.

File: MANIFEST.in
Comment (babolivier) [reply MANIFEST.in:6]: As per #synapse-dev, it looks like `python -m build` always includes all files from the repository by default when run on my machine, not sure why.

According to https://packaging.python.org/en/latest/guides/using-manifest-in/, `pyproject.toml` is automatically included now. I've also added the license file to `license_file` in `setup.cfg` since it seems like that will always cause it to be included.

Comment (babolivier) [reply MANIFEST.in:6]: I've now replaced the content of `MANIFEST.in` with the suggestion from `check-manifest`. The only exception to this is that I've added a `prune .github` to remove GitHub codeowner and workflow files, which felt to me like they shouldn't live in the sdist.

File: MANIFEST.in
Comment (babolivier) [reply MANIFEST.in:2]: That's for `releasing.md`. The readme is apparently automatically included according to https://packaging.python.org/en/latest/guides/using-manifest-in/.

**Review by DMRobertson — APPROVED** [review]
Thanks, SGTM. Are you okay to make a (patch) release with these changes?

**Comment by babolivier** [comment]
> Are you okay to make a (patch) release with these changes?

Yep that was my plan once this is merged (I need the changes released for something else anyway).

**Pull request merged**
